plugins/ipfs/browser/browseripfs.go

```diff
@@ -0,0 +1,415 @@
+}
+
+func (l *BrowserIpfs) RunCmd(ctx context.Context, stdin io.Reader, args ...string) (testbedi.Output, error) {
+	env, err := l.env()
+
+	if err != nil {
+		return nil, fmt.Errorf("error getting env: %s", err)
+	}
+
+	cmd := exec.CommandContext(ctx, args[0], args[1:]...)
+	cmd.Env = env
+	cmd.Stdin = stdin
+
+	stderr, err := cmd.StderrPipe()
+	if err != nil {
+		return nil, err
+	}
+
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return nil, err
+	}
+
+	err = cmd.Start()
+	if err != nil {
+		return nil, err
+	}
+
+	stderrbytes, err := ioutil.ReadAll(stderr)
+	if err != nil {
+		return nil, err
+	}
+
+	stdoutbytes, err := ioutil.ReadAll(stdout)
+	if err != nil {
+		return nil, err
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	exiterr := cmd.Wait()
+
+	var exitcode = 0
+	switch oerr := exiterr.(type) {
+	case *exec.ExitError:
+		if ctx.Err() == context.DeadlineExceeded {
+			err = errors.Wrapf(oerr, "context deadline exceeded for command: %q", strings.Join(cmd.Args, " "))
+		}
+
+		exitcode = 1
+	case nil:
+		err = oerr
+	}
+
+	return iptbutil.NewOutput(args, stdoutbytes, stderrbytes, exitcode, err), nil
+}
+
+func (l *BrowserIpfs) Connect(ctx context.Context, tbn testbedi.Core) error {
+	swarmaddrs, err := tbn.SwarmAddrs()
+	if err != nil {
+		return err
+	}
+
+	for _, addr := range swarmaddrs {
+		output, err := l.RunCmd(ctx, nil, "ipfs", "swarm", "connect", addr)
+
+		if err != nil {
+			return err
+		}
+
+		if output.ExitCode() == 0 {
+			return nil
+		}
+	}
+
+	return fmt.Errorf("Could not connect using any address")
+}
+
+func (l *BrowserIpfs) Shell(ctx context.Context, nodes []testbedi.Core) error {
+	shell := os.Getenv("SHELL")
+	if shell == "" {
+		return fmt.Errorf("no shell found")
+	}
+
+	if len(os.Getenv("IPFS_PATH")) != 0 {
+		// If the users shell sets IPFS_PATH, it will just be overridden by the shell again
+		return fmt.Errorf("shell has IPFS_PATH set, please unset before trying to use iptb shell")
+	}
+
+	nenvs, err := l.env()
+	if err != nil {
+		return err
+	}
+
+	// TODO(tperson): It would be great if we could guarantee that the shell
+	// is using the same binary. However, the users shell may prepend anything
+	// we change in the PATH
+
+	for i, n := range nodes {
+		peerid, err := n.PeerID()
+
+		if err != nil {
+			return err
+		}
+
+		nenvs = append(nenvs, fmt.Sprintf("NODE%d=%s", i, peerid))
+	}
+
+	return syscall.Exec(shell, []string{shell}, nenvs)
+}
+
+func (l *BrowserIpfs) String() string {
+	pcid, err := l.PeerID()
+	if err != nil {
+		return fmt.Sprintf("%s", l.Type())
+	}
+	return fmt.Sprintf("%s", pcid[0:12])
+}
+
+func (l *BrowserIpfs) APIAddr() (string, error) {
+	return ipfs.GetAPIAddrFromRepo(l.dir)
+}
+
+func (l *BrowserIpfs) SwarmAddrs() ([]string, error) {
+	return ipfs.SwarmAddrs(l)
+}
+
+func (l *BrowserIpfs) Dir() string {
+	return l.dir
+}
+
+func (l *BrowserIpfs) PeerID() (string, error) {
+	if l.peerid != nil {
+		return l.peerid.String(), nil
+	}
+
+	var err error
```

Comment:
Do we need to do this?

Reply:
Ya, can't use `:=` as `l.peerid` is already define.